/**:
  ros__parameters:
    crosswalk:
      common:
        show_processing_time: false # [-] whether to show processing time
        # param for input data
        traffic_light_state_timeout: 3.0 # [s] timeout threshold for traffic light signal
        enable_rtc: true # if true, the scene modules should be approved by (request to cooperate)rtc function. if false, the module can be run without approval from rtc.

      # param for stop position
      stop_position:
        stop_position_threshold: 1.0 # [m] threshold to check whether the vehicle stops in front of crosswalk for yielding

        # For the Lanelet2 map with no explicit stop lines
        stop_distance_from_crosswalk: 3.5 # [m] make stop line away from crosswalk
        # For the case where the crosswalk width is very wide
        far_object_threshold: 10.0 # [m] If objects cross X meters behind the stop line, the stop position is determined according to the object position (stop_distance_from_object meters before the object).
        # For the case where the stop position is determined according to the object position.
        stop_distance_from_object: 2.0 # [m] the vehicle decelerates to be able to stop in front of object with margin

      # param for ego's slow down velocity
      slow_down:
        min_slow_down_velocity: 2.78 # [m/s] target vehicle velocity when module receive slow down command from FOA (2.78 m/s = 10.0 kmph)
        max_slow_down_jerk: -1.5 # [m/sss] minimum jerk deceleration for safe brake
        max_slow_down_accel: -2.5 # [m/ss] minimum accel deceleration for safe brake
        no_relax_velocity: 2.78 # [m/s] if the current velocity is less than X m/s, ego always stops at the stop position(not relax deceleration constraints 2.78 m/s = 10 kmph)

      # param for stuck vehicles
      stuck_vehicle:
        stuck_vehicle_velocity: 1.0 # [m/s] maximum velocity threshold whether the vehicle is stuck
        max_stuck_vehicle_lateral_offset: 2.0 # [m] maximum lateral offset for stuck vehicle position should be looked
        stuck_vehicle_attention_range: 10.0 # [m] the detection area is defined as X meters behind the crosswalk
        min_acc: -1.0   # min acceleration [m/ss]
        min_jerk: -1.0  # min jerk [m/sss]
        max_jerk: 1.0   # max jerk [m/sss]

      # param for pass judge logic
      pass_judge:
        ego_pass_first_margin_x: [0.0] # [[s]] time to collision margin vector for ego pass first situation (the module judges that ego don't have to stop at TTC + MARGIN < TTV condition)
        ego_pass_first_margin_y: [6.0] # [[s]] time to vehicle margin vector for ego pass first situation (the module judges that ego don't have to stop at TTC + MARGIN < TTV condition)
        ego_pass_first_additional_margin: 0.5 # [s] additional time margin for ego pass first situation to suppress chattering
        ego_pass_later_margin_x: [0.0] # [[s]] time to vehicle margin vector for object pass first situation (the module judges that ego don't have to stop at TTV + MARGIN < TTC condition)
        ego_pass_later_margin_y: [10.0] # [[s]] time to collision margin vector for object pass first situation (the module judges that ego don't have to stop at TTV + MARGIN < TTC condition)
        ego_pass_later_additional_margin: 0.5 # [s] additional time margin for object pass first situation to suppress chattering

        no_stop_decision:
          max_offset_to_crosswalk_for_yield: 0.0 # [m] maximum offset from ego's front to crosswalk for yield. Positive value means in front of the crosswalk.
          min_acc: -1.0   # min acceleration [m/ss]
          min_jerk: -1.0  # min jerk [m/sss]
          max_jerk: 1.0   # max jerk [m/sss]

        stop_object_velocity_threshold: 0.28 # [m/s] velocity threshold for the module to judge whether the objects is stopped (0.28 m/s = 1.0 kmph)
        min_object_velocity: 1.39 # [m/s] minimum object velocity (compare the estimated velocity by perception module with this parameter and adopt the larger one to calculate TTV. 1.39 m/s = 5.0 kmph)
        ## param for yielding
        disable_stop_for_yield_cancel: true # for the crosswalk where there is a traffic signal
        disable_yield_for_new_stopped_object: true # for the crosswalk where there is a traffic signal
        # if the pedestrian does not move for X seconds after stopping before the crosswalk, the module judge that ego is able to pass first.
        distance_map_for_no_intention_to_walk: [1.0, 5.0] # [m] ascending order
        timeout_map_for_no_intention_to_walk: [3.0, 0.0] # [s]
        timeout_ego_stop_for_yield: 3.0 # [s] the amount of time which ego should be stopping to query whether it yields or not

      # param for target object filtering
      object_filtering:
        crosswalk_attention_range: 1.0 # [m] the detection area is defined as -X meters before the crosswalk to +X meters behind the crosswalk
        target_object:
          unknown: true # [-] whether to look and stop by UNKNOWN objects
          bicycle: true # [-] whether to look and stop by BICYCLE objects
          motorcycle: true # [-] whether to look and stop by MOTORCYCLE objects (tmp: currently it is difficult for perception modules to detect bicycles and motorcycles separately.)
          pedestrian: true # [-] whether to look and stop by PEDESTRIAN objects
